### PR TITLE
Bug 1923843 - New ping: dau-reporting 

### PR DESCRIPTION
**Pull request by badboy**
This change allows us to offer Firefox users more granular choices in data collection.

### CODE REVIEW

**Comment by badboy** [comment]
Data review request, cc @travis79:

1. What questions will you answer with this data?

We'd like a dedicated ping to measure DAU.
This change allows us to offer Firefox users more granular choices in data collection.

2. Why does Mozilla need to answer these questions? Are there benefits for users?
   Do we need this information to address product or business requirements?

This is a product requirement, so we better understand how Firefox is used.

3. What alternative methods did you consider to answer these questions?
   Why were they not sufficient?

Relying on existing data.
We cannot split those out from other data collection,
so a dedicated ping is the better way forward.

4. Can current instrumentation answer these questions?

As above. Not independently from other data collection.

5. List all proposed measurements and indicate the category of data collection for each
   measurement, using the Firefox data collection categories found on the Mozilla wiki.

Measurement Name | Measurement Description | Data Collection Category | Tracking Bug
---------------- | ----------------------- | ------------------------ | ------------
`browser_search.ad_clicks` | Records clicks of adverts on SERP pages. The key format is `<provider-name>`.  |  | https://bugzilla.mozilla.org/show_bug.cgi?id=1923843
`search.counts` | The labels for this counter are `{search-engine-name}.{source}`  If the search engine is bundled with Firefox-iOS, then `search-engine-name` will be the name of the search engine. If it is a custom search engine, the value will be `custom`.  The value of `source` will reflect the source from which the search started.  One of: * quicksearch * suggestion * actionbar  |  | https://bugzilla.mozilla.org/show_bug.cgi?id=1923843
`search.default_engine` | The default search engine identifier if the search engine is pre-loaded with Firefox-iOS.  If it's a custom search engine, then the value will be 'custom'.  |  | https://bugzilla.mozilla.org/show_bug.cgi?id=1923843
`search.in_content` | Records the type of interaction a user has on SERP pages.  |  | https://bugzilla.mozilla.org/show_bug.cgi?id=1923843

6. Please provide a link to the documentation for this data collection which
   describes the ultimate data set in a public, complete, and accurate way.

This collection is Glean so is documented [in the Glean Dictionary](https://dictionary.telemetry.mozilla.org).

7. How long will this data be collected?

The new ping will be collected with no expiry.
Some of the metrics have an expiry '2025-01-01'.

8. What populations will you measure?

All channels, countries, and locales. No filters.

9. If this data collection is default on, what is the opt-out mechanism for users?

These collections are Glean. The opt-out can be found in the product's preferences.

10. Please provide a general description of how you will analyze this data.

This will be used to gather overall DAU, presented in our dashboards.

11. Where do you intend to share the results of your analysis?

In Looker dashboards.

12. Is there a third-party tool (i.e. not Glean or Telemetry) that you
    are proposing to use for this data collection?

No.

**Review by travis79 — APPROVED** [review]
> Data review request, cc @travis79:
> 
>     1. What questions will you answer with this data?
> 
> 
> We'd like a dedicated ping to measure DAU. This change allows us to offer Firefox users more granular choices in data collection.
> 
>     2. Why does Mozilla need to answer these questions? Are there benefits for users?
>        Do we need this information to address product or business requirements?
> 
> 
> This is a product requirement, so we better understand how Firefox is used.
> 
>     3. What alternative methods did you consider to answer these questions?
>        Why were they not sufficient?
> 
> 
> Relying on existing data. We cannot split those out from other data collection, so a dedicated ping is the better way forward.
> 
>     4. Can current instrumentation answer these questions?
> 
> 
> As above. Not independently from other data collection.
> 
>     5. List all proposed measurements and indicate the category of data collection for each
>        measurement, using the Firefox data collection categories found on the Mozilla wiki.
> 
> 
> Measurement Name 	Measurement Description 	Data Collection Category 	Tracking Bug
> `browser_search.ad_clicks` 	Records clicks of adverts on SERP pages. The key format is `<provider-name>`. 		https://bugzilla.mozilla.org/show_bug.cgi?id=1923843
> `search.counts` 	The labels for this counter are `{search-engine-name}.{source}` If the search engine is bundled with Firefox-iOS, then `search-engine-name` will be the name of the search engine. If it is a custom search engine, the value will be `custom`. The value of `source` will reflect the source from which the search started. One of: * quicksearch * suggestion * actionbar 		https://bugzilla.mozilla.org/show_bug.cgi?id=1923843
> `search.default_engine` 	The default search engine identifier if the search engine is pre-loaded with Firefox-iOS. If it's a custom search engine, then the value will be 'custom'. 		https://bugzilla.mozilla.org/show_bug.cgi?id=1923843
> `search.in_content` 	Records the type of interaction a user has on SERP pages. 		https://bugzilla.mozilla.org/show_bug.cgi?id=1923843
> 
>     6. Please provide a link to the documentation for this data collection which
>        describes the ultimate data set in a public, complete, and accurate way.
> 
> 
> This collection is Glean so is documented [in the Glean Dictionary](https://dictionary.telemetry.mozilla.org).
> 
>     7. How long will this data be collected?
> 
> 
> The new ping will be collected with no expiry. Some of the metrics have an expiry '2025-01-01'.
> 
>     8. What populations will you measure?
> 
> 
> All channels, countries, and locales. No filters.
> 
>     9. If this data collection is default on, what is the opt-out mechanism for users?
> 
> 
> These collections are Glean. The opt-out can be found in the product's preferences.
> 
>     10. Please provide a general description of how you will analyze this data.
> 
> 
> This will be used to gather overall DAU, presented in our dashboards.
> 
>     11. Where do you intend to share the results of your analysis?
> 
> 
> In Looker dashboards.
> 
>     12. Is there a third-party tool (i.e. not Glean or Telemetry) that you
>         are proposing to use for this data collection?
> 
> 
> No.

# Data Review

1) Is there or will there be **documentation** that describes the schema for the ultimate data set in a public, complete, and accurate way?

Yes, through the metrics.yaml file and the [Glean Dictionary](https://dictionary.telemetry.mozilla.org/).

2) Is there a control mechanism that allows the user to turn the data collection on and off?

Yes, through the data preferences in the application settings.

3) If the request is for permanent data collection, is there someone who will monitor the data over time?

All metric collection expiry dates will remain the same as before, sending the data on a different cadence doesn't change when the specific metrics expire.

4) Using the **[category system of data types](https://wiki.mozilla.org/Data_Collection)** on the Mozilla wiki, what collection type of data do the requested measurements fall under?

All sensitive and non-sensitive data has previously been reviewed and approved

5) Is the data collection request for default-on or default-off?

Default-on

6) Does the instrumentation include the addition of **any *new* identifiers** (whether anonymous or otherwise; e.g., username, random IDs, etc.  See the appendix for more details)?

No

7) Is the data collection covered by the existing Firefox privacy notice?

Yes

8) Does the data collection use a third-party collection tool?

No

## Result

data-review+

File: firefox-ios/Client/Telemetry/TelemetryWrapper.swift [hunk 1]
```diff
@@ -104,8 +104,13 @@ class TelemetryWrapper: TelemetryWrapperProtocol, FeatureFlaggable {
             GleanMetrics.LegacyIds.clientId.set(uuid)
         }
 
+        glean.registerPings(GleanMetrics.Pings.shared)
+
         // Initialize Glean telemetry
-        let gleanConfig = Configuration(channel: AppConstants.buildChannel.rawValue, logLevel: .off)
```
Comment (OrlaM) [inline firefox-ios/Client/Telemetry/TelemetryWrapper.swift:108]: Is logLevel being deprecated? Will Glean logs show up in the console again or is it just defaulted to off?

Comment (badboy) [reply firefox-ios/Client/Telemetry/TelemetryWrapper.swift:108]: Woops, thanks for catching that. I removed it for local testing, and then promptly forgot to put it back.

**Review by OrlaM — APPROVED** [review]

**Comment by OrlaM** [comment]
@MergifyIO backport release/v132

**Comment by OrlaM** [comment]
@MergifyIO backport release/v132

**Pull request merged**